src/provider.ts

```diff
@@ -12,52 +12,12 @@ import {
   StandardResolutionReasons,
 } from '@openfeature/server-sdk';
 
+import type { Evaluation, HyphenEvaluationContext, HyphenProviderOptions } from './types';
 import pkg from '../package.json';
-import NodeCache from '@cacheable/node-cache';
+import { HyphenClient } from './hyphenClient';
 
-export type clientContext = {
-  targetingKey: string;
-  ipAddress?: string;
-  customAttributes?: Record<string, any>;
-  user?: {
-    id: string;
-    email?: string;
-    name?: string;
-    customAttributes?: Record<string, any>;
-  };
-} & EvaluationContext;
-
-export interface HyphenEvaluationContext extends EvaluationContext {
-  targetingKey: string;
-  ipAddress: string;
-  application: string;
-  environment: string;
-  customAttributes: Record<string, any>;
-  user: {
-    id: string;
-    email: string;
-    name: string;
-    customAttributes: Record<string, any>;
-  };
-}
-
-export interface Evaluation {
-  key: string;
-  value: boolean | string | number | Record<string, any>;
-  type: 'boolean' | 'string' | 'number' | 'object';
-  reason: string;
-  errorMessage: string;
-}
-
-export interface EvaluationResponse {
-  toggles: Record<string, Evaluation>;
-}
-
-export class HyphenProvider implements Provider {
-  private readonly publicKey: string;
-  private readonly application: string;
-  private readonly environment: string;
-  private cache: NodeCache;
+export class HyphenProvider extends HyphenClient implements Provider {
```

Comment:
this should not inherit from the client it should just consume the client. Inheriting is the wrong abstraction.